feat: wsh `export-config` command

weilirs opened the pull request:
feat #1608 
`export-config [output-path]` to export config files into a zip archive.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/wavetermdev/waveterm?pullRequest=1641) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/wavetermdev/waveterm?pullRequest=1641) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/wavetermdev/waveterm?pullRequest=1641) it.</sub>

esimkowitz commented on cmd/wsh/cmd/wshcmd-export-config.go:
```diff
@@ -0,0 +1,88 @@
+}
+
+func runExportConfig(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("export-config requires an output path")
+	}
+
+	outputPath := args[0]
+	if !strings.HasSuffix(outputPath, ".zip") {
+		outputPath += ".zip"
+	}
+
+	configDir := getWaveConfigDir()
+	if err := zipDir(configDir, outputPath); err != nil {
+		return fmt.Errorf("export-config failed: %v", err)
+	}
+
+	fmt.Printf("Configuration exported to %s\n", outputPath)
+	return nil
+}
```
@weilirs this shouldn't ever happen but it's probably good for posterity. Could you check that the config dir path is not empty before walking?

weilirs commented:
I added a check for the existence of config directory, please take a look

esimkowitz commented:
Looks good to me! Can you add documentation to the `wsh-reference.mdx` file for your new command?

weilirs commented:
> Looks good to me! Can you add documentation to the `wsh-reference.mdx` file for your new command?

I added to the documentation and I modified the command from `export-config` to `exportconfig` for consistency.

esimkowitz commented on cmd/wsh/cmd/wshcmd-export-config.go:
```diff
@@ -0,0 +1,93 @@
+	})
+}
+
+func getWaveConfigDir() string {
```
This is not reliable as the user may have a different config directory defined, such as on dev where it's `~/.config/waveterm-dev`. You should use `wavebase.GetWaveConfigDir` instead.

esimkowitz replied:
However, because this is called from the wsh side, you should just call the `wshclient.PathCommand` function, since that will get you whatever the value is in wavesrv

esimkowitz replied:
actually, this won't work at all if you're on a remote machine. The logic for zipping the config should be moved to wavesrv and called through the wshserver mechanism, you can look at the other commands to see how we do this. Then, if the user is invoking this command from a remote machine, we should copy the zipped archive to the output path on the remote machine

weilirs replied:
To copy the zipped archive to the output path on the remote machine, should I use the `filestore.WFS.WriteFile` or something else? I am using `filestore.WFS.WriteFile` but `ls` doesn't show anything on the remote machine. thanks!

esimkowitz replied:
I'm working on a more flexible filesystem traversal mechanism right now, targeting end of this week. Once this is implemented, I'll let you know so you can integrate it here. For now, let's hold on this PR

esimkowitz commented on docs/docs/wsh-reference.mdx:
````diff
@@ -153,6 +153,16 @@ wsh editconfig presets/ai.json
 
 ---
 
+## exportconfig
+
+You can export your Wave's config files into a zip archive using this command.
+
+```
+wsh exportconfig [output-path]
+```
+
+---
````
good suggestion @weilirs

sawka commented:
closed due to age
